src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -315,16 +315,25 @@ def refreshCache(self):
 
                 # all files in job have same location (in se names)
                 rawLocations = loadedJob["input_files"][0]["locations"]
+                
+                sitedb = SiteDB()
+ #               sitedb_names = set()
+                for l in rawLocations:
+#                    sitedb_names.update(sitedb.PNNtoPSN(l) or [])
+                    possibleLocations.update(sitedb.PNNtoPSN(l) or [])
```

Comment:
Once we rename wmbs_location_senames  to wmbs_location_pnn then this will only be the pnn. I think that we still need to map that to the PSN at this point right?

--Alex
________________________________
From: ticoann [notifications@github.com]
Sent: 22 July 2015 16:01
To: dmwm/WMCore
Cc: Richards, Alexander J
Subject: Re: [WMCore] PNN transition combined patch (Ready to be merged) (#5989)


In src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py<https://github.com/dmwm/WMCore/pull/5989#discussion_r35222549>:

> @@ -315,16 +315,25 @@ def refreshCache(self):
>
>                  # all files in job have same location (in se names)
>                  rawLocations = loadedJob["input_files"][0]["locations"]
> +
> +                sitedb = SiteDB()
> + #               sitedb_names = set()
> +                for l in rawLocations:
> +#                    sitedb_names.update(sitedb.PNNtoPSN(l) or [])
> +                    possibleLocations.update(sitedb.PNNtoPSN(l) or [])


One problem getting the mapping from sitedb at this point, if sitedb goes down temporary or there is network problem.
The job submission will fail. I think we should get this information from wmbs_location_senames table instead.
We can create the memory cache as well we don't create call db every time.

—
Reply to this email directly or view it on GitHub<https://github.com/dmwm/WMCore/pull/5989/files#r35222549>.